src/providers/tiktok.ts

```diff
@@ -0,0 +1,73 @@
+import { custom } from "openid-client"
+import type { TokenSet, TokenSetParameters } from "openid-client"
+import { Profile } from ".."
+import { OAuthConfig, OAuthUserConfig } from "./oauth"
+
+export interface TikTokProfile extends Profile {
+  avatar: string
+  open_id: string
+  display_name: string
+}
+
+export default function TikTok<P extends TikTokProfile>(
+  options: OAuthUserConfig
+): OAuthConfig<P> {
+  return {
+    id: "tiktok",
+    name: "TikTok",
+    type: "oauth",
+    version: "2.0",
+    authorization: {
+      params: {
+        scope: "user.info.basic",
+        client_key: options.clientId,
+      },
+      url: "https://open-api.tiktok.com/platform/oauth/connect/?response_type=code",
+    },
+    token: {
+      url: "https://open-api.tiktok.com/oauth/access_token/",
+      request: async ({ checks, client, params, provider }) => {
+        const data = await client.grant({
+          code: params.code,
+          redirect_uri: provider.callbackUrl,
+          params,
+          grant_type: "authorization_code",
+          client_key: options.clientId,
+          client_secret: options.clientSecret,
+          code_verifier: checks.code_verifier,
+        })
+        const tokens = data.data as TokenSetParameters
+
+        return { tokens }
+      },
+    },
+    userinfo: {
+      url: "https://open-api.tiktok.com/oauth/userinfo/",
+      request: async ({ client, tokens }) => {
+        client[custom.http_options] = (options) => {
+          return { ...options, followRedirect: true, timeout: 99999 }
```

Comment:
There is now a top-level option for setting timeout:

https://github.com/nextauthjs/next-auth/pull/3188

I would also suggest setting something more sensible, like 5-10 seconds.